actors/evm/src/interpreter/precompiles.rs

```diff
@@ -57,79 +78,145 @@ impl<BS: Blockstore, RT: Runtime<BS>> Precompiles<BS, RT> {
     }
 }
 
-/// read 32 bytes (u256) from buffer or error
-fn read_u256(buf: &[u8], start: usize) -> Result<U256, PrecompileError> {
-    let slice = buf.get(start..start + 32).ok_or(PrecompileError::IncorrectInputSize)?;
-    Ok(U256::from_big_endian(slice))
+/// Intermediary container to hold a data as slice or turn into a vec with padding as needed.
+/// This is intended to not copy in normal operations
+enum Data<'a> {
+    Slice(&'a [u8]),
+    Vec(Vec<u8>),
```

Comment:
Cow? We don't have any methods defined on this anyways, so we could cover all this with a:

```rust
fn right_pad(data: impl Into<Cow<[u8]>) -> Cow<[u8]> { ... }
```